include/xtensor/xreducer.hpp

```diff
@@ -384,22 +397,25 @@ namespace xt
      //@{
      /**
       * Constructs an xreducer expression applying the specified
-      * function to the given expression over the given axis.
+      * function to the given expression over the given axes.
       *
       * @param func the function to apply
       * @param e the expression to reduce
-      * @param axis the axis along which the reduction is performed
+      * @param axes the axes along which the reduction is performed
       */
     template <class F, class CT, class X>
-    template <class Func>
-    inline xreducer<F, CT, X>::xreducer(Func&& func, CT e, const axis_storage& axis)
-        : m_e(e), m_f(std::forward<Func>(func)), m_axis(axis),
-          m_shape(make_sequence<shape_type>(m_e.dimension() - axis.size(), 0)),
+    template <class Func, class AX>
+    inline xreducer<F, CT, X>::xreducer(Func&& func, CT e, AX&& axes)
+        : m_e(e), m_f(std::forward<Func>(func)), m_axes(axes),
+          m_shape(make_sequence<shape_type>(m_e.dimension() - axes.size(), 0)),
```

Comment:
If you forward ``axes``, you should use to ``m_axes.size()`` instead of ``axes.size()`` to avoid dangling reference when ``axes`` is moved. Note the standard ensure ``m_axes`` will be initialized before ``m_shape`` since it appears before in the class definition.